File: Message/MessageType.cs
namespace SharpTS.Message
{
	/// <summary>
	/// Message type enum
	/// </summary>
	public enum MessageType
	{
		/// <summary>
		/// Send from client to server, notifying about DOMContentLoaded
		/// </summary>
		DOMContentLoaded = 0,
		
		/// <summary>
		/// Load another component
		/// </summary>
		Load,
		
		/// <summary>
		/// Sync state of ViewModel
		/// </summary>
		SyncState,

		/// <summary>
		/// Fetch data
		/// </summary>
		Fetch,
		
		/// <summary>
		/// Notify about new frame
		/// </summary>
		NewFrame
	}
}

File: Client/message/MessageType.ts
export enum MessageType {
	/**
	 * Send notification about DOMContentLoaded
	 */
	DOMContentLoaded,
	
	/**
	 * Load another component
	 */
	Load,

	/**
	 * Sync state with C# component ViewModel
	 */
	SyncState,

	/**
	 * Fetch data
	 */
	Fetch,

	/**
	 * Notify C# about new frame
	 */
	NewFrame
}